### DISCUSSION TITLE
Allow user-configurable upload descriptions

### DISCUSSION THREAD

**Discussion by windexvalence** [RogueOneEcho/caesura#101] [Ideas]
While the auto-generated upload descriptions are great, there are certain scenarios where it may be useful for a user to override the standard auto-genned description for one or more of their uploads.

Obviously, these description overrides can't be as dynamic as the current auto-genned descriptions, but I was thinking that something as simple as an option pointing to a description file might work well as a start. Whenever the option is not present, the standard auto-generated description is used as normal. For example, something like the following:

```shell
docker compose run --rm caesura upload <source-here> --description-file /some/local_path/to_my/description.txt
```
Where the provided `description.txt` could be simply a text file with non-templated bbcode text (i.e. something like the example below):
```txt
My manual description which I'm using instead of the default caesura description.
```

**Comment by RogueOneEcho** [comment]
In what scenario would you want an alternative fixed or static description?

At present, `caesura` only uploads transcodes from a source FLAC so in my mind documenting the source is a fairly essential feature.

For example, it helps with moderation, if the source turns out to contain a watermark or other issue then the transcodes from that source will also need to be identified and removed.

Descriptions for MP3s also contain eye3d output giving clear insight into the tracks which should help with moderation.

**Reply by RogueOneEcho** [reply]
Closing and marking as won't fix as I don't see the value in this and haven't received a response.

Feel free to suggest reasons for it to be re-opened

**Reply by windexvalence** [reply]
Forgot to respond here. I hear you and generally agree. The  main reason I was thinking about description modification was for adding lineage details for cassette / vinyl releases. I'm happy to make that a separate discussion topic if you think that is worthwhile. Otherwise, I understand the reasoning for closing this, and thanks for the detailed response.

**Reply by RogueOneEcho** [reply]
Lineage will either be:

### 1. a file in the source torrent 

Since [v0.23.0](https://github.com/RogueOneEcho/caesura/releases/tag/v0.23.0) `.cue`,  `.log`,  `.nfo`,  `.rtf`, and `.txt` files are now included in transcodes so the lineage will be included with the transcode. Therefore I don't see any value in editing the description as well.

### 2. included in the description of the source torrent

The link to the source torrent is included in the description of the torrent which *in my opinion* is sufficient to explain the lineage. Staff disagree with this as there is a possibility the source torrent will be deleted and therefore the lineage lost. *My opinion* is that if the source is deleted then most likely the transcode should be deleted too as any issue with the source will be an issue with the transcode, therefore it doesn't matter that the lineage isn't explicitly in the description.

If users do want to explicitly add the lineage to the description then it's definitely going to be simpler for them to manually edit through the site than to create a `.txt` file or pass CLI args etc.

**Comment by RogueOneEcho** [comment]
Another user has requested the ability to remove the description footer:

Therefore I'm going to re-open this discussion and suggest implementing it as follows:

- [ ] Add a `description_footer` option to `config.yml` that overrides the existing footer message.

Example:

```yaml
description_footer: |
  Hello, world!
  This is a multi-line footer message added to description that can include [b]BB code[/b].
```

If `null` it will use the default footer:

https://github.com/RogueOneEcho/caesura/blob/9eb6ab2ed053a3684dd5a0481c30bda64b73b025/src/commands/upload/upload_command.rs#L285-L288

```yaml
description_footer: null
```

If empty it will disable the default footer:

```yaml
description_footer: ''
```
